flow.py

```diff
@@ -57,17 +68,85 @@ def dest(self):
     def dest(self, dest):
         raise AttributError("Cannot modify a flow's destination node.")
 
-    def make_packet(self, id, src, dest, data):
+    def check_last(self):
+        """Method that checks if all packets have been acknowledged and
+        updating the NetworkSimulator object accordingly
+        """
+        if len(self.unacknowledged) == 0:
+            self.ns.decrement_active_flows()
+
+    def update_flow(a_packet):
+        """Upon receiving an acknowledgement packet, updating the flow's
+        attributes
+
+        Args:
+            a_packet (AcknowledgementPacket): Packet being sent
+                back from host
+        """
+        print("Flow: updated with acknowledgement packet: ", a_packet.id,
+            "And flow", self.flow_id)
```

Comment:
use the same format for this print statement as some of the other ones you have below.